packages/client/hmi-client/src/page/project/components/code-editor.vue

```diff
@@ -476,22 +167,28 @@ async function initialize(editorInstance) {
 }
 
 async function createModelFromCode() {
+	createModelLoading.value = true;
 	if (selectedPaper.value) {
-		// createModelLoading.value = true;
-		// const response = await API.post(`/code/annotation/find_text_vars?text=${mockPaperText}`);
-		// createModelLoading.value = false;
-	}
-	const modelName = 'New model';
-	const newModel = {
-		name: 'New model',
-		framework: 'Petri Net',
-		content: JSON.stringify(acset.value)
-	};
-	const model = await createModel(newModel);
-	console.log(model);
-	if (model) {
-		emit('on-model-created', model.id, modelName);
+		const paperToExtractMetadata = await getDocumentById(selectedPaper.value[0].xdd_uri);
+		if (paperToExtractMetadata) {
+			const response = await API.post(
+				`/code/annotation/find_text_vars?text=${paperToExtractMetadata.abstract}`
+			);
+			const newModelName = 'New model';
+			const newModel = {
+				name: newModelName,
+				framework: 'Petri Net',
+				content: JSON.stringify({ ...acset.value, ...response.data })
+			};
+			const model = await createModel(newModel);
+			if (model) {
+				emit('on-model-created', model.id, newModelName);
+			} else {
+				logger.error(`Something went wrong.`);
+			}
```

Comment:
This part doesn't look right, we created a model but no project information had been submitted that I can see. Were do we associate the model with the project we are on?

Reply:
oh I see the project-asset stuff is done outside .... kind of weird, would prefer to have the transactional actions in one place, but okay for now.